File: src/net/jimmc/mimprint/AreaLayoutFactory.java
/* AreaLayoutFactory.java
 *
 * Jim McBeath, Nov 7, 2005
 */

package net.jimmc.jiviewer;

public class AreaLayoutFactory {
    //TODO use a singleton of this class and make the method not static
    /** Create an empty AreaLayout of the specifed type. */
    public static AreaLayout newAreaLayout(String type) {
        if (type.equals("imageLayout"))
            return new ImagePageArea();
        else if (type.equals("gridLayout"))
            return new AreaGridLayout();
        else if (type.equals("splitLayout"))
            return new AreaSplitLayout();
        else
            throw new IllegalArgumentException("Unknown area type "+type); //TODO i18n
    }
}

File: src/net/jimmc/mimprint/AreaLayoutFactory.scala
/* AreaLayoutFactory.java
 *
 * Jim McBeath, Nov 7, 2005
 * Converted to scala June 21, 2008
 */

package net.jimmc.mimprint

object AreaLayoutFactory {
    /** Create an empty AreaLayout of the specifed type. */
    def newAreaLayout(layoutType:String):AreaLayout = {
        layoutType match {
            case "imageLayout" => new AreaImageLayout()
            case "gridLayout"  => new AreaGridLayout()
            case "splitLayout" => new AreaSplitLayout()
            case _ =>
                throw new IllegalArgumentException(
                        "Unknown area type "+layoutType); //TODO i18n
        }
    }

    def createDefaultTopLayout():AreaLayout = new AreaImageLayout(0,0,0,0)
}
